python/lsst/ts/MTMount/mock/controller.py

```diff
@@ -239,16 +252,25 @@ async def put_camera_cable_wrap_telemetry(self, tai):
         )
         await self.write_telemetry(data_dict)
 
-    async def write_reply(self, reply):
+    async def write_reply(self, reply_dict):
         """Write a reply to the command/reply stream.
+
+        Parameters
+        ----------
+        reply_dict : `dict`
+            Reply as a dict.
+            It will be formatted as json before being written.
         """
-        self.log.debug("write_reply(%s)", reply)
-        self.command_server.writer.write(reply.encode())
+        self.log.debug("write_reply(%s)", reply_dict)
```

Comment:
Can you please use preffered f"xx" formatting? So f"write_reply({reply})"? Also true for other % usages in controller.py

Reply:
I normally agree but not for expensive debug log messages, such as this.

With the %s formatting and the parameters sent separately the formatting is only done if the log level is >= DEBUG, which it is not by default.